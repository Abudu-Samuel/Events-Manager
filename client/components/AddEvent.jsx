import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import Navbar from '../components/Navbar';
import Footer from '../components/Footer';

import * as userActions from '../actions/actionCreator';

/**
 * 
 * 
 * @class AddEvent
 * @extends {React.Component}
 */
class AddEvent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      time: '',
      date: '',
      description: '',
      type: '',
      image: '',
      errorStatus: false,
      redirect: false,
    };
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = event => {
    event.preventDefault();
    this.setState({
      errorMessage: '',
      errorStatus: false
    });
    this.props.addEvent({ ...this.state, centerId: this.props.match.params.centerId })
    .then(() => {
      this.setState({
        errorStatus: false,
        errorMessage: ''
      });
      setTimeout(() => {
        this.setState({
          redirect: true
        });
      }, 1000);
    })
    .catch((error) => {
      this.setState({
        errorMessage: error.response.data.message,
        errorStatus: true
      })
    })
    
  }

  render() {
    return (
      this.state.redirect ?
          <Redirect to={`/centers/${this.props.match.params.centerId}`} /> :
      <div className="space">
        <Navbar />
        <div className="container add">
          <section>
            <h4 className="font-weight-bold text-center">Create An Event</h4>
          </section>
          <section>
            <div className="container">
              <form onSubmit={this.handleSubmit} className="signup add">
                <div className="md-form">
                  <i className="fa fa-ticket prefix teal-text" />
                  <input type="text" name="title" onChange={this.handleChange} id="first-name" className="form-control" />
                  <label htmlFor="orangeForm-name">Event Title</label>
                </div>
                <div className="md-form">
                  <i className="fa fa-clock-o prefix teal-text" />
                  <input type="text" name="date" onChange={this.handleChange} id="username-name" className="form-control" />
                  <label htmlFor="orangeForm-name">Event Date</label>
                </div>
                <div className="md-form">
                  <i className="fa fa-clock-o prefix teal-text" />
                  <input type="text" name="time" onChange={this.handleChange} id="ends-name" className="form-control" />
                  <label htmlFor="orangeForm-name">Event Time</label>
                </div>
                <div className="md-form">
                  <i className="fa fa-clock-o prefix teal-text" />
                  <input type="text" name="type" onChange={this.handleChange} id="ends-name" className="form-control" />
                  <label htmlFor="orangeForm-name">Event Type</label>
                </div>
                <div className="md-form myfile">
                  <i className="fa fa-camera-retro prefix teal-text" />
                  <input type="text" id="image" name="image" className="form-control" onChange={this.handleChange} />
                                <label htmlFor="orangeForm-name">Image</label>
                </div>
                <div className="md-form form-sm">
                  <i className="fa fa-pencil prefix teal-text" />
                  <textarea type="text" name="description" onChange={this.handleChange} className="md-textarea" id="input-4" />
                  <label htmlFor="input-4">Event Description</label>
                </div>
                {
                            this.state.errorStatus ?
                              <h5 className="text-center font-weight-bold red-text">{this.state.errorMessage}</h5> :
                              null
                          }
                <div className="text-center">
                            <button className="btn btn-mycolor mb-3">Add Center<i className="fa fa-sign-in ml-1" /></button>
                          </div>
              </form>
            </div>
          </section>
        </div>
        <Footer />
      </div>
    );
  }
}

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} dispatch
 *
 * @return {object} mapped dispatch
 */
function mapDispatchToProps(dispatch) {
  return {
    addEvent: (eventData) => dispatch(userActions.addEvent(eventData))
  };
}

export default connect(null, mapDispatchToProps)(AddEvent);
