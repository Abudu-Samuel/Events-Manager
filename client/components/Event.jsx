import React from 'react';

const Event = ({ event }) => ((
  <div className="row">
    <div className="col-xl-4 mb-4">
      <div className="card">
        <img className="img-fluid" src={event.image} alt="Card image cap" />
        <div className="card-body">
          <p className="grey-text">{event.date}</p>
          <h4 className="card-title pt-1 text-center">{event.title}</h4>
        </div>
      </div>
    </div>
    <div className="col-xl-8">
      <h5 className="mb-2 font-weight-bold grey-text">The coming of Kings</h5>
      <p className="grey-text">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo adipisicing
                                      elit, sed adipioris nisi.
      </p>
      <p className="grey-text">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo adipisicing
                                      elit, sed adipim ad minim veniam, quis nostrud exercitation ullamco laboris nisi.
      </p>
      <p className="grey-text">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo adipisicing
                                      elit, sed adipim ad minim veniam, quis nostrud exercitation ullamco laboris nisi.
      </p>
      <a className="btn btn-default btn-sm">Read More</a>
    </div>
  </div>
));

export default Event;
