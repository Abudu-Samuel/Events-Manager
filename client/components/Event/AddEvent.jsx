import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import axios from 'axios';
import Navbar from '../common/Navbar';
import Footer from '../common/Footer';
import Form from '../common/forms/Form';
import { validateEvent } from '../Utils/Validator';

import * as userActions from '../../actions/actionCreator';

/**
 *
 *
 * @class AddEvent
 * @extends {React.Component}
 */
class AddEvent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      date: '',
      description: '',
      type: '',
      image: '',
      imgPreviewSrc: '',
      errorStatus: false,
      errors: {},
      redirect: false,
      editing: false
    };
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleUpload = (event) => {
    event.preventDefault();
    const file = event.target.files[0];
    let imgPreview = document.getElementById('img-preview');
    let imgPreviewSrc = imgPreview.src
    const CLOUDINARY_URL = 'https://api.cloudinary.com/v1_1/leumas/upload';
    const CLOUDINARY_UPLOAD_PRESET = 'cf3etily';
    const formData = new FormData();
    formData.append('file', file);
    formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);

    axios({
      url: CLOUDINARY_URL,
      method: 'POST',
      headers: {
        'Content-Type': 'application/x-www- form-urlencoded '
      },
      data: formData
    })
      .then((res) => {
        this.setState({
          image: res.data.secure_url,
          imgPreviewSrc: res.data.secure_url
        });
      })
      .catch((err) => {
        throw err
      });
  }

  handleSubmit = event => {
    event.preventDefault();
    const validationErrors = validateEvent(this.state).errors;
    if (Object.keys(validationErrors).length > 0) {
      this.setState({ errors: validationErrors });
      return;
    }
    this.setState({
      errorMessage: '',
      errorStatus: false
    });
    this.props.addEvent({ ...this.state, centerId: this.props.match.params.centerId })
      .then(() => {
        this.setState({
          errorStatus: false,
          errorMessage: ''
        });
        setTimeout(() => {
          this.setState({
            redirect: true
          });
        }, 1000);
      })
      .catch((error) => {
        this.setState({
          errorMessage: error.response.data.message,
          errorStatus: true
        });
      });
  }

  render() {
    return (
      this.state.redirect ?
        <Redirect to={`/centers/${this.props.match.params.centerId}`} /> :
        <div className="space">
          <Navbar />
          <div className="container add">
            <section>
              <h4 className="font-weight-bold text-center">Create An Event</h4>
            </section>
            <section>
              <div className="container">
                <Form
                  handleSubmit={this.handleSubmit}
                  handleChange={this.handleChange}
                  handleUpload={this.handleUpload}
                  errorStatus={this.state.errorStatus}
                  errors={this.state.errors}
                  errorMessage={this.state.errorMessage}
                  editing={this.state.editing}
                  title={this.state.title}
                  date={this.state.date}
                  description={this.state.description}
                  type={this.state.type}
                  image={this.state.image}
                  imgPreviewSrc={this.state.imgPreviewSrc}
                />
              </div>
            </section>
          </div>
          <Footer />
        </div>
    );
  }
}

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} dispatch
 *
 * @return {object} mapped dispatch
 */
function mapDispatchToProps(dispatch) {
  return {
    addEvent: (eventData) => dispatch(userActions.addEvent(eventData))
  };
}

export default connect(null, mapDispatchToProps)(AddEvent);
