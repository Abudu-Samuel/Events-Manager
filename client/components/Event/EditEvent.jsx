import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import axios from 'axios';
import Navbar from '../common/Navbar';
import Form from '../common/forms/Form';
import { validateEvent } from '../Utils/Validator';
import * as userActions from '../../actions/actionCreator';


class EditEvent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      date: '',
      description: '',
      type: '',
      image: '',
      errorStatus: false,
      redirect: false,
      editing: false,
      imgPreviewSrc: '',
      errors: {},
      errorMessage: '',
      centerId: ''
    };
  }

  componentDidMount() {
    this.props.singleEvent(this.props.match.params.eventId);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.getSingleEvent) {
      this.setState({
        title: nextProps.getSingleEvent.event.title,
        centerId: nextProps.getSingleEvent.event.centerId,
        date: nextProps.getSingleEvent.event.date,
        description: nextProps.getSingleEvent.event.description,
        type: nextProps.getSingleEvent.event.type,
        image: nextProps.getSingleEvent.event.image,
      });
    }
  }
  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

handleSubmit = (event) => {
  const eventId = this.props.match.params.eventId;
  event.preventDefault();
  const validationErrors = validateEvent(this.state).errors;
  if (Object.keys(validationErrors).length > 0) {
    this.setState({ errors: validationErrors });
    return;
  }
  this.setState({
    errorMessage: '',
    errorStatus: false
  });
  this.props.editEvent({ data: this.state, eventId: eventId })
    .then(() => {
      this.setState({
        errorStatus: false,
        errorMessage: ''
      });
      setTimeout(() => {
        this.setState({
          redirect: true
        });
      }, 100);
    })
    .catch((error) => {
      this.setState({
        errorMessage: error.response.data.message,
        errorStatus: true
      });
    });
}

handleUpload = (event) => {
  event.preventDefault();
  const file = event.target.files[0];
  let imgPreview = document.getElementById('img-preview');
  let imgPreviewSrc = imgPreview.src
  const CLOUDINARY_URL = 'https://api.cloudinary.com/v1_1/leumas/upload';
  const CLOUDINARY_UPLOAD_PRESET = 'cf3etily';
  const formData = new FormData();
  formData.append('file', file);
  formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);

  axios({
    url: CLOUDINARY_URL,
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www- form-urlencoded '
    },
    data: formData
  })
    .then((res) => {
      this.setState({
        image: res.data.secure_url,
        imgPreviewSrc: res.data.secure_url
      });
    })
    .catch((err) => {
      throw err
    });
}

render() {
  console.log('........', this.state.image)
  return (
    this.state.redirect ?
      <Redirect to="/manage/events" /> :
      <div className="space">
        <Navbar />
        <div className="container add">
          <section>
            <h4 className="font-weight-bold text-center">Edit Event</h4>
          </section>
          <section>
            <div className="container">
              <Form
                editing={this.state.editing}
                errorStatus={this.state.errorStatus}
                errorMessage={this.state.errorMessage}
                title={this.state.title}
                errors={this.state.errors}
                handleUpload={this.handleUpload}
                date={this.state.date}
                description={this.state.description}
                type={this.state.type}
                image={this.state.image}
                imgPreviewSrc={this.state.imgPreviewSrc}                
                handleChange={this.handleChange}
                handleSubmit={this.handleSubmit}/>
            </div>
          </section>
        </div>
      </div>
  );
}
}

function mapStateToProps(state, ownProps) {
  return {
    getSingleEvent: state.events.event
  };
}

function mapDispatchToProps(dispatch) {
  return {
    singleEvent: eventData => dispatch(userActions.singleEvent(eventData)),
    editEvent: eventData => dispatch(userActions.editEvent(eventData))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditEvent);
