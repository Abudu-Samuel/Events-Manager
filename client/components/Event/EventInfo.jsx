import React from 'react';


const EventDetails = ({
  event,
  noEvent
}) => (
  <div>
    <div className="space">
      <div className="container" style={{ marginTop: 99 }}>
        <h2 className="mb-5 font-weight-bold  text-center">{event.title}</h2>
        <div className="row">
          <div className="col-xl-6 mb-4">
            <div className="card prev">
              <img className="img-fluid equal" src={event.image} alt="Card image cap" />
            </div>
          </div>
          <div className="col-xl-6 text-justify">
            <div className="row mb-5">
              <div className="col-xl-4">
                <p className="mb-2 font-weight-bold ">{event.title}</p>
              </div>
              <div className="col-xl-4">
                <p className="font-weight-bold">Type: {event.type}</p>
              </div>
              <div className="col-xl-4">
                <h6 className="font-weight-bold">Date: {event.date}</h6>
              </div>
            </div>
            <p className="text-center font-weight-bold">Description</p>
            <p className="font-weight-bold">{event.description}</p>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default EventDetails;
