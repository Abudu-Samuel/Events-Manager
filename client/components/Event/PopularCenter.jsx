import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const PopularEvents = ({
  events,
  isAuthenticated,
  getEventId
}) => (
  <div className="container space">
    <section id="events">
      <h2 className="mb-3 slot text-center font-weight-bold">{
        isAuthenticated ? 'Events' : 'Latest Events'
      }</h2>
      <div className="row mb-4">
        {
          events.event.map(event => (
            <div className="col-md-4 mb-4" key={event.id}>
              <div className="card hoverable text-center">
                <img className="img-fluid hoverable max" src={event.image} />
                <div className="card-body">
                  <h4 className="card-title"> {
                    event.title.split('').length > 17 ?
                      `${event.title.slice(0, 27)}...` : event.title
                  }</h4>
                  <p className="card-text">
                    {
                      event.description.split('').length > 40 ?
                        `${event.description.slice(0, 40)}...` :
                        event.description
                    }
                  </p>
                  <p className="card-text">Date: {event.date}</p>
                  <button
                    id="viewEvent"
                    className="btn btn-mycolor btn-sm"
                    data-centerid={event.id}
                    onClick={getEventId}>
                    <Link to={`/events/${event.id}`}>Details</Link>
                  </button>
                </div>
              </div>
            </div>
          ))
        }
      </div>
    </section>
  </div>
);

PopularEvents.propTypes = {
  events: PropTypes.object,
  isAuthenticated: PropTypes.bool,
  getEventId: PropTypes.func
};

export default PopularEvents;
