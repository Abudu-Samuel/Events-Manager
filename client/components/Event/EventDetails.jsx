import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { BounceLoader } from 'react-spinners';
import Navbar from '../common/Navbar';
import EventInfo from './EventInfo';
import * as userActions from '../../actions/actionCreator';

/**
 * @class EventDetails
 *
 * @extends {React.Component}
 */
export class EventDetails extends React.Component {
  /**
   * Creates an instance of EventDetails.
   *
   * @param {any} props
   *
   * @memberof EventDetails
   */
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      event: {},
      noEvent: false,
      fetchingEvent: false,
      loading: true,
      fetchingCenter: false
    };
  }

  /**
   * @method componentDidMount
   *
   * @description React life cycle
   *
   * @returns {object} state
   *
   * @memberof EventDetails
   */
  componentDidMount() {
    this.props.singleEvent(this.props.match.params.eventId);
  }

  /**
   * @description React life cycle
   *
   * @returns {object} updated state
   *
   * @param {nextProps} nextProps
   *
   * @memberof EventDetails
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.event) {
      this.setState({ event: nextProps.event.event, fetchingEvent: false, loading: false });
    } else {
      this.setState({ fetchingEvent: true });
    }
  }

  /**
   * @method render
   *
   * @description React method render
   *
   * @returns {jsx} Jsx representation of the dom
   *
   * @memberof EventDetails
   */
  render() {
    return (
      <div>
        <Navbar />
        <BounceLoader
          color={'#123abc'}
          loading={this.state.loading}
        />
        <EventInfo
          event={this.state.event}
          noEvent={this.state.noEvent}
        />
      </div>
    );
  }
}

EventDetails.propTypes = {
  singleEvent: PropTypes.func,
  event: PropTypes.object,
  match: PropTypes.object
};

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} state
 *
 * @param {function} ownProps
 *
 * @return {object} mapped dispatch
 */
const mapStateToProps = state => ({
  event: state.events.event,
  getSingleCenter: state.centers.center
});

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} dispatch
 *
 * @return {object} mapped dispatch
 */
export const mapDispatchToProps = dispatch => bindActionCreators({
  singleEvent: eventData => userActions.singleEvent(eventData),
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(EventDetails);
