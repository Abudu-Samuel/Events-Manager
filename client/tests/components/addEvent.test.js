import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { AddEvent, mapDispatchToProps } from '../../components/Event/AddEvent';


const event = {
  preventDefault: jest.fn(),
  handleUpload: () => Promise.resolve(),
  target: {
    name: 'test',
    files: [{ data: 'samuel', type: 'image/jpg' }]
  }
};

describe('<AddEvent />', () => {
  it('should render Footer correctly', () => {
    const wrapper = shallow(<AddEvent />);
    wrapper.instance().handleChange(event);
    wrapper.instance().handleSubmit(event);
    wrapper.instance().handleUpload(event);
    return Promise.resolve().then(() => {
      wrapper.setProps({
        image: 'img',
        imgPreviewSrc: 'mg'
      });
      expect(event.preventDefault).toHaveBeenCalled();
      expect(wrapper).toMatchSnapshot();
      expect();
    });
  });
});

describe('the mapDispatchToProps function', () => {
  it('should return an object of functions to be passed as props', () => {
    const dispatchFn = jest.fn();
    const propsObj = mapDispatchToProps(dispatchFn);

    expect(propsObj).toEqual({
      addEvent: expect.any(Function),
    });

    propsObj.addEvent();

    expect(dispatchFn).toHaveBeenCalledTimes(1);
  });
});
