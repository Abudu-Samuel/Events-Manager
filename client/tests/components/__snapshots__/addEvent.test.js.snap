// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddEvent /> should render Footer correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddEvent
    image="img"
    imgPreviewSrc="mg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <div
          className="container add"
        >
          <section>
            <h4
              className="font-weight-bold text-center"
            >
              Create An Event
            </h4>
          </section>
          <section>
            <div
              className="container"
            >
              <Form
                date=""
                description=""
                editing={false}
                errorStatus={false}
                errors={
                  Object {
                    "date": "Date is required",
                    "description": "Description should contain alphabet/number",
                    "image": "Image is Required",
                    "title": "Event title should contain alphabet/number",
                    "type": "Type should contain only alphabets",
                  }
                }
                handleChange={[Function]}
                handleSubmit={[Function]}
                handleUpload={[Function]}
                image=""
                imgPreviewSrc=""
                loading={true}
                title=""
                type=""
              />
            </div>
          </section>
        </div>,
        <Footer />,
      ],
      "className": "space",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section>
              <h4
                className="font-weight-bold text-center"
              >
                Create An Event
              </h4>
            </section>,
            <section>
              <div
                className="container"
              >
                <Form
                  date=""
                  description=""
                  editing={false}
                  errorStatus={false}
                  errors={
                    Object {
                      "date": "Date is required",
                      "description": "Description should contain alphabet/number",
                      "image": "Image is Required",
                      "title": "Event title should contain alphabet/number",
                      "type": "Type should contain only alphabets",
                    }
                  }
                  handleChange={[Function]}
                  handleSubmit={[Function]}
                  handleUpload={[Function]}
                  image=""
                  imgPreviewSrc=""
                  loading={true}
                  title=""
                  type=""
                />
              </div>
            </section>,
          ],
          "className": "container add",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4
                className="font-weight-bold text-center"
              >
                Create An Event
              </h4>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create An Event",
                "className": "font-weight-bold text-center",
              },
              "ref": null,
              "rendered": "Create An Event",
              "type": "h4",
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
              >
                <Form
                  date=""
                  description=""
                  editing={false}
                  errorStatus={false}
                  errors={
                    Object {
                      "date": "Date is required",
                      "description": "Description should contain alphabet/number",
                      "image": "Image is Required",
                      "title": "Event title should contain alphabet/number",
                      "type": "Type should contain only alphabets",
                    }
                  }
                  handleChange={[Function]}
                  handleSubmit={[Function]}
                  handleUpload={[Function]}
                  image=""
                  imgPreviewSrc=""
                  loading={true}
                  title=""
                  type=""
                />
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Form
                  date=""
                  description=""
                  editing={false}
                  errorStatus={false}
                  errors={
                    Object {
                      "date": "Date is required",
                      "description": "Description should contain alphabet/number",
                      "image": "Image is Required",
                      "title": "Event title should contain alphabet/number",
                      "type": "Type should contain only alphabets",
                    }
                  }
                  handleChange={[Function]}
                  handleSubmit={[Function]}
                  handleUpload={[Function]}
                  image=""
                  imgPreviewSrc=""
                  loading={true}
                  title=""
                  type=""
                />,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "date": "",
                  "description": "",
                  "editing": false,
                  "errorMessage": undefined,
                  "errorStatus": false,
                  "errors": Object {
                    "date": "Date is required",
                    "description": "Description should contain alphabet/number",
                    "image": "Image is Required",
                    "title": "Event title should contain alphabet/number",
                    "type": "Type should contain only alphabets",
                  },
                  "handleChange": [Function],
                  "handleSubmit": [Function],
                  "handleUpload": [Function],
                  "image": "",
                  "imgPreviewSrc": "",
                  "loading": true,
                  "title": "",
                  "type": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "section",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <div
            className="container add"
          >
            <section>
              <h4
                className="font-weight-bold text-center"
              >
                Create An Event
              </h4>
            </section>
            <section>
              <div
                className="container"
              >
                <Form
                  date=""
                  description=""
                  editing={false}
                  errorStatus={false}
                  errors={
                    Object {
                      "date": "Date is required",
                      "description": "Description should contain alphabet/number",
                      "image": "Image is Required",
                      "title": "Event title should contain alphabet/number",
                      "type": "Type should contain only alphabets",
                    }
                  }
                  handleChange={[Function]}
                  handleSubmit={[Function]}
                  handleUpload={[Function]}
                  image=""
                  imgPreviewSrc=""
                  loading={true}
                  title=""
                  type=""
                />
              </div>
            </section>
          </div>,
          <Footer />,
        ],
        "className": "space",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <section>
                <h4
                  className="font-weight-bold text-center"
                >
                  Create An Event
                </h4>
              </section>,
              <section>
                <div
                  className="container"
                >
                  <Form
                    date=""
                    description=""
                    editing={false}
                    errorStatus={false}
                    errors={
                      Object {
                        "date": "Date is required",
                        "description": "Description should contain alphabet/number",
                        "image": "Image is Required",
                        "title": "Event title should contain alphabet/number",
                        "type": "Type should contain only alphabets",
                      }
                    }
                    handleChange={[Function]}
                    handleSubmit={[Function]}
                    handleUpload={[Function]}
                    image=""
                    imgPreviewSrc=""
                    loading={true}
                    title=""
                    type=""
                  />
                </div>
              </section>,
            ],
            "className": "container add",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h4
                  className="font-weight-bold text-center"
                >
                  Create An Event
                </h4>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create An Event",
                  "className": "font-weight-bold text-center",
                },
                "ref": null,
                "rendered": "Create An Event",
                "type": "h4",
              },
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="container"
                >
                  <Form
                    date=""
                    description=""
                    editing={false}
                    errorStatus={false}
                    errors={
                      Object {
                        "date": "Date is required",
                        "description": "Description should contain alphabet/number",
                        "image": "Image is Required",
                        "title": "Event title should contain alphabet/number",
                        "type": "Type should contain only alphabets",
                      }
                    }
                    handleChange={[Function]}
                    handleSubmit={[Function]}
                    handleUpload={[Function]}
                    image=""
                    imgPreviewSrc=""
                    loading={true}
                    title=""
                    type=""
                  />
                </div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Form
                    date=""
                    description=""
                    editing={false}
                    errorStatus={false}
                    errors={
                      Object {
                        "date": "Date is required",
                        "description": "Description should contain alphabet/number",
                        "image": "Image is Required",
                        "title": "Event title should contain alphabet/number",
                        "type": "Type should contain only alphabets",
                      }
                    }
                    handleChange={[Function]}
                    handleSubmit={[Function]}
                    handleUpload={[Function]}
                    image=""
                    imgPreviewSrc=""
                    loading={true}
                    title=""
                    type=""
                  />,
                  "className": "container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "date": "",
                    "description": "",
                    "editing": false,
                    "errorMessage": undefined,
                    "errorStatus": false,
                    "errors": Object {
                      "date": "Date is required",
                      "description": "Description should contain alphabet/number",
                      "image": "Image is Required",
                      "title": "Event title should contain alphabet/number",
                      "type": "Type should contain only alphabets",
                    },
                    "handleChange": [Function],
                    "handleSubmit": [Function],
                    "handleUpload": [Function],
                    "image": "",
                    "imgPreviewSrc": "",
                    "loading": true,
                    "title": "",
                    "type": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "section",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
